pkg/platform/runtime/runtime.go

```diff
@@ -100,42 +100,33 @@ func (r *Runtime) Target() setup.Targeter {
 
 // Update updates the runtime by downloading all necessary artifacts from the Platform and installing them locally.
 // This function is usually called, after New() returned with a NeedsUpdateError
-func (r *Runtime) Update(auth *authentication.Auth, msgHandler *events.RuntimeEventHandler) error {
+func (r *Runtime) Update(auth *authentication.Auth, eventHandler events.Handler) (rerr error) {
 	if r.disabled {
 		return nil // nothing to do
 	}
 
 	logging.Debug("Updating %s#%s @ %s", r.target.Name(), r.target.CommitUUID(), r.target.Dir())
 
 	// Run the setup function (the one that produces runtime events) in the background...
-	prod := events.NewRuntimeEventProducer()
-	var setupErr error
-	go func() {
-		defer func() {
-			r.recordCompletion(setupErr)
-			prod.Close()
-		}()
-
-		if err := setup.New(r.target, prod, auth, r.analytics).Update(); err != nil {
-			setupErr = errs.Wrap(err, "Update failed")
-			return
+	defer func() {
+		r.recordCompletion(rerr)
+		if err := eventHandler.Close(); err != nil {
```

Comment:
Ideally we should not be closing this here as this package was not responsible for constructing it.